src/writer/app_runner.py

```
@@ -691,6 +693,73 @@ def on_any_event(self, event) -> None:
+        self._log_hash()
+
+    def _calculate_project_hash(self) -> str:
+        project_hash = hashlib.md5()
+        for filename in sorted(self.wf_project_context.file_hashes.keys()):
+            file_hash = self.wf_project_context.file_hashes[filename]
+            project_hash.update(bytes.fromhex(file_hash))
+        return project_hash.hexdigest()
+
+    def _process_file(self, file_path) -> None:
+        try:
+            file_hash = hashlib.md5()
```

Ruff (0.14.10)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)